gwcs/wcs/_wcs.py

```diff
@@ -1175,16 +1144,49 @@ def transform(
         to_step = self._get_step(to_frame)
         transform = self.get_transform(from_step.step.frame, to_step.step.frame)
 
-        if not transform.uses_quantity and is_high_level(
-            *args, low_level_wcs=from_step.step.frame
-        ):
-            args = high_level_objects_to_values(
-                *args, low_level_wcs=from_step.step.frame
-            )
+        with_bounding_box = kwargs.get("with_bounding_box")
+        fill_value = kwargs.get("fill_value", np.nan)
```

Comment:
these should be added as kw-only arguments with default values in the signature.